### PR TITLE
 remove final troublesome deps from SSCS

### PR DESCRIPTION
**Pull request by deads2k**
Builds on a previous pull.  We'll move the content of pkg/controller into library-go (and perhaps ultimately into kube).  this removes the last non-portable links to kube/kube and openshift/orign.  After updating library-go and vendoring back here, it should be possible to move the SSCS to openshift/service-serving-cert-signer.

/assign @liggitt 

@openshift/sig-security fyi.  I don't think there's anything contentious in the pull

### CODE REVIEW

**Comment by mfojtik** [comment]
LGTM

File: pkg/service/controller/servingcert/secret_creating_controller.go [hunk 1]
```diff
@@ -162,7 +161,7 @@ func (sc *ServiceServingCertController) enqueueService(obj interface{}) {
 	if !ok {
 		return
 	}
-	key, err := controller.KeyFunc(obj)
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
```
Comment (mfojtik) [inline pkg/service/controller/servingcert/secret_creating_controller.go]: nit: the cast to `v1.Service` above will prevent check of `obj == DeletedFinalStateUnknown` right? So you can just use meta.MetaNamespaceKeyFunc

Comment (deads2k) [reply pkg/service/controller/servingcert/secret_creating_controller.go]: > nit: the cast to v1.Service above will prevent check of obj == DeletedFinalStateUnknown right? So you can just use meta.MetaNamespaceKeyFunc

This one is the fully portable version.  I'd like to keep using everywhere we can.

Comment (mfojtik) [reply pkg/service/controller/servingcert/secret_creating_controller.go]: k, i guess we also want to fix the image import scheduler https://github.com/openshift/origin/pull/19687

**Pull request merged**